### ISSUE TITLE
"Xlib.error.DisplayNameError: Bad display name" - do I need XQuartz?

### ISSUE THREAD

**Issue by baltasvejas** [JeffHoogland/pyxhook#2]
Thanks for the software. I hope it will work on mac OS X as well.
I had to install Xlib and I done this on my virtualenv with 

    pip install svn+https://svn.code.sf.net/p/python-xlib/code/trunk/

Now I get with pip freeze

    python-xlib==0.15rc1

When running the script I get the error:

    (env)@mymac:/mytest/keylog$ python keylog.py 
    Traceback (most recent call last):
      File "keylog.py", line 21, in <module>
        hookman = pyxhook.HookManager()
      File "/mytest/keylog/pyxhook.py", line 83, in __init__
        self.local_dpy = display.Display()
      File "/mytest/keylog/env/lib/python2.7/site-packages/Xlib/display.py", line 83, in __init__
        self.display = _BaseDisplay(display)
      File "/mytest/keylog/env/lib/python2.7/site-packages/Xlib/display.py", line 65, in __init__
        apply(protocol.display.Display.__init__, (self, ) + args, keys)
      File "/mytest/keylog/env/lib/python2.7/site-packages/Xlib/protocol/display.py", line 45, in __init__
        name, host, displayno, screenno = connect.get_display(display)
      File "/mytest/keylog/env/lib/python2.7/site-packages/Xlib/support/connect.py", line 65, in get_display
        return mod.get_display(display)
      File "/mytest/keylog/env/lib/python2.7/site-packages/Xlib/support/unix_connect.py", line 59, in get_display
        raise error.DisplayNameError(display)
    Xlib.error.DisplayNameError: Bad display name ""
    (env)@mymac:/mytest/keylog$ 

It looks like the problem is that DISPLAY environment variable is empty:

    (env)@mymac:/mytest/keylog$ echo $DISPLAY
    
    (env)@mymac:/mytest/keylog$ 

Does this mean I need to install XQuartz?
Or there is a problem with Xlib?

**Comment by JeffHoogland** [comment]
Not sure - never worked on OSX sorry.

**Comment by baltasvejas** [comment]
It appears I have to install `XQuartz`. 

After installing it, I ran into the issue with `$DISPLAY` variable (can be seen in [this post](http://apple.stackexchange.com/a/171654/152981)).

And finally I stuck with `RECORD extension not found`, because QXuartz doesn't support RECORD extension. [This post](http://stackoverflow.com/a/28184402/2647009) explains that RECORD extension is not suitable for working with inputs on OS X.

The positive side of this, is that I learned some stuff about X on OS X and at some time I will try pyxhook on linux.

Thanks
